external-crates/move/crates/move-package/src/compilation/compiled_package.rs

```diff
@@ -452,12 +457,16 @@ impl CompiledPackage {
             &resolved_package,
             transitive_dependencies,
         )?;
+
+        let (deps_for_current_compiler, deps_for_prior_compiler) =
+            partition_deps_by_toolchain(deps_package_paths.clone())?;
+
         let flags = resolution_graph.build_options.compiler_flags();
         // invoke the compiler
-        let mut paths = deps_package_paths.clone();
+        let mut paths = deps_for_current_compiler;
         paths.push(sources_package_paths.clone());
 
-        let compiler = Compiler::from_package_paths(paths, vec![])
+        let compiler = Compiler::from_package_paths(paths, deps_for_prior_compiler)
```

Comment:
Request for input: this is the "main" change. In effect, we compile deps whose `Move.lock` specify a prior compiler version other than the one we're on. For those deps, we pass their package paths in the second arg to `from_package_paths`, which is one route I found that _appears_ to work well and completes the build.

One change in the `build` directory output I observe though, is that, prior to this change, the `build` directory would include dependency `.mv` files in `build/dependencies`. With this change, `.mv` files of dependencies compiled with a prior compiler do not end up in `build/dependencies` (and I'm not sure they need to?).

Reply:
Is it possible to test this change? That is, add a new Move project that uses Move 2024, but relies on a Move Legacy version of code that is already published? This would allow use to ensure the change supports on-chain verification while building correctly.

Reply:
Also, I think moving the `.mv` files to `build/dependencies/` is a good and worthwhile goal if we can. it will allow us to cache dependencies by-version, etc., in the future.

Reply:
> Is it possible to test this change? That is, add a new Move project that uses Move 2024, but relies on a Move Legacy version of code that is already published? This would allow use to ensure the change supports on-chain verification while building correctly.

I tested locally to get a successful build, and will check on-chain verification with a proper PR. This PR was meant to get a sense of how to do the build, which I think is ~mostly clear to me now.

> Also, I think moving the .mv files to build/dependencies/ is a good and worthwhile goal if we can. it will allow us to cache dependencies by-version, etc., in the future.

Right, so this is the kind of thing that I was a bit unsure about re how the internals work when `build/dependencies` exist. I.e., if we copy / move `.mv` to `build/dependencies` would caching "just work"?